app/Http/Controllers/ContestsController.php

```diff
@@ -134,4 +138,41 @@ public function export(Contest $contest)
         $csv = $contest->getCSVExport();
         $csv->output('contest' . $contest->id . '.csv');
     }
+
+    /**
+     * Grab the form to add a user retroactively.
+     *
+     * @param  \Gladiator\Models\Contest $contest
+     * @return  \Illuminate\Http\Response
+     */
+    public function signupForm(Contest $contest)
+    {
+        $contest = $contest->load(['waitingroom', 'competitions']);
+
+        return view('contests.signup', compact('contest'));
+    }
+
+    /**
+     * The post request to add a user retroactively.
+     *
+     * @param  SignupUserRequest $request
+     * @return \Illuminate\Http\Response
+     */
+    public function signupUser(SignupUserRequest $request)
+    {
+        $user = $this->registrar->findUserAccount($request->all());
+
+        if (! $user instanceof User) {
+            $user = $this->registrar->createUser($user);
+        }
+
+        // Add the user to a competition or waiting room, but only if not already in it.
+        if ($request->has('competition_id') && ! $user->competitions()->where('id', $request->competition_id)->first()) {
```

Comment:
So you could do something like:

```php

public function addUserToCompetition($user, $competitionId)
{
  $user = $user->competitions()->where('id', $competition_id)->first();

  if ($user) {
    throw new UserExistsInCompetitionException;
  } else {
    // add user and return user to calling function.
  }
}
```

then you can catch the exception in the calling function.  If this feels too over the top, maybe just leave what you had before. This could be useful if we foresee using these methods in other scenarios in the future.